straxen/get_corrections.py

```diff
@@ -34,152 +36,75 @@ def correction_options_wrapped(run_id, conf, *arg):
-            #   (
-            #   "to_pe_model", -> This is to specify that we want  the gains
-            #   "v1", -> The version of the correction 'v1' is the online version
-            #   )
-            raise ValueError('CMT gain model should be similar to:'
-                             '("CMT_model", ("to_pe_model", "v1"). Instead got:'
-                             f'{model_conf}')
-        # is this the best way to do this?
-        is_nt = n_pmts == straxen.n_tpc_pmts or n_pmts == straxen.n_nveto_pmts or n_pmts == straxen.n_mveto_pmts
-
-        corrections = straxen.CorrectionsManagementServices(is_nt=is_nt)
-        to_pe = corrections.get_corrections_config(run_id, model_conf)
-
-        return to_pe
+def get_correction_from_cmt(run_id, conf):
+    """
+    Get correction from CMT general format is
+    conf = ('correction_name', 'version', True)
+    where True means looking at nT runs, e.g. 
+    get_correction_from_cmt(run_id, conf[:2])
+    special cases:
+    version can be replaced by consant int, float or array
+    when user specify value(s)
+    :param run_id: run id from runDB
+    :param conf: configuration 
+    :return: correction value(s)
+    """
+ 
+    if isinstance(conf, str) and conf.startswith('https://raw'):
+        # Legacy support for pax files
+        return conf
 
-    elif model_type == 'to_pe_per_run':
-        warn("to_pe_per_run will be replaced by CorrectionsManagementSevices",
-             DeprecationWarning, 2)
-        # Load a npy file specifying a run_id -> to_pe array
-        to_pe_file = model_conf
-        x = straxen.get_resource(to_pe_file, fmt='npy')
-        run_index = np.where(x['run_id'] == int(run_id))[0]
-        if not len(run_index):
-            # Gains not known: using placeholders
-            run_index = [-1]
-        to_pe = x[run_index[0]]['to_pe']
+    elif isinstance(conf, tuple) and len(conf) == 2:
+        model_conf, cte_value = conf[:2]
 
-    elif model_type == 'to_pe_constant':
+        # special case constant to_pe values 
         if model_conf in FIXED_TO_PE:
-            return FIXED_TO_PE[model_conf]
-
-        try:
-            # Uniform gain, specified as a to_pe factor
-            to_pe = np.ones(n_pmts, dtype=np.float32) * model_conf
-        except np.core._exceptions.UFuncTypeError as e:
-            raise (str(e) +
-                   f"\nTried multiplying by {model_conf}. Insert a number instead.")
-    else:
-        raise NotImplementedError(f"Gain model type {model_type} not implemented")
+            correction = FIXED_TO_PE[model_conf]
+            return correction
 
-    if len(to_pe) != n_pmts:
-        raise ValueError(
-            f"Gain model {gain_model} resulted in a to_pe "
-            f"of length {len(to_pe)}, but n_pmts is {n_pmts}!")
-    return to_pe
+        # special case constant single value
+        elif 'constant' in model_conf:
+            if not isinstance(cte_value, (float, int)):
```

Comment:
Right, but it will be nice to pass only the file name instead of the full path, and then the mongo downloader can take care of the rest. The same concept also applies to [the correction files](https://github.com/ahiguera-mx/straxen/blob/ab7b4c485e1270b12302196e8c89ba0e4adbe473/straxen/plugins/event_processing.py#L459-L460), I believe.

Reply:
I see what you mean something like 

```
if isintance(conf, str):
      downloader = straxen.MongoDownloader()
      downloader.download_single(conf)
      return path_to_the_file
```

But at then end you need to pass the path to the file, right?

Reply:
Wait, I think you can do what you one already since `get_resources()` can handle a few options [here](https://github.com/ahiguera-mx/straxen/blob/master/straxen/common.py#L152)